packages/jason-api/src/types/request.ts

```diff
@@ -40,9 +40,9 @@ interface Transformer<DataOut = any> {
     (response: any): ResponseShape<DataOut>;
 }
 
-type SetRelationshipOnSuccess = [string, string, string, FlexiblePayload];
+type SetRelationshipOnSuccess = [string, string, string, FlexiblePayload | null | undefined];
```

Comment:
The documentation for this functionality (which is out-of-date in other ways 😟), actually expects the last piece of the tuple to be omitted if not used, so I think it would be best if the type reflected that:

```suggestion
type SetRelationshipOnSuccess =
    | [string, string, string]
    | [string, string, string, FlexiblePayload];
```